app/Repositories/PhotoRepository.php

```diff
@@ -44,7 +44,11 @@ public function create(array $data, $signupId)
     {
         $postEvent = Event::create($data);
 
-        $fileUrl = $this->aws->storeImage($data['file'], $data['campaign_id']);
+        if (array_key_exists('campaign_id', $data)) {
+            $fileUrl = $this->aws->storeImage($data['file'], $data['campaign_id']);
+        } else {
+            $fileUrl = $this->aws->storeImage($data['file'], $signupId);
+        }
```

Comment:
i don't think so just pass the `signupId` into the `crop` function or set it as a property of the class and then you can use it however you want.